### ISSUE TITLE
Password in Windows credentials is not updated on next `set_password` call

### ISSUE THREAD

**Issue by wl2776** [jaraco/keyring#545]
**Describe the bug**
New entry is created instead of updating existing password.

**To Reproduce**

1. Open cmd.exe and set password with CLI utility
    ```
    C:\>keyring set service username
    Password for 'username' in 'service': <enter the password>
    ```

2. Open Credential manager and observe a new entry:
    ![image](https://user-images.githubusercontent.com/4774284/142363885-08a7dc78-7e8e-413e-bd1f-20db1173cbc5.png)

3. Update password. Open cmd.exe again and set new password with CLI utility for the same service and user name
    ```
    C:\>keyring set service username
    Password for 'username' in 'service': <enter another password>
    ```

4. Refresh Credential manager's window and see two entries: `service` and `username@service`

    ![image](https://user-images.githubusercontent.com/4774284/142364254-8b5b29a9-a5f9-4fa8-a508-35e8f6d3b44e.png)

**Expected behavior**
Second call for `keyring set service username` should modify existing entry instead of creating new one.

**Environment**

- OS: Windows 10 Pro with all recent updates.

```
C:\> pip list | grep keyring
keyring                    23.2.1

C:\> keyring --list-backends
keyring.backends.Windows.WinVaultKeyring (priority: 5)
keyring.backends.chainer.ChainerBackend (priority: -1)
keyring.backends.fail.Keyring (priority: 0)
```

**Additional context**
The same behavior is observed in call to `keyring.set_password` in python scripts.

**Comment by jrobbins-LiveData** [comment]
I've done some work on the Windows backend  (see pull request #544) that, I believe, fixes this issue.

**Comment by jaraco** [comment]
That's interesting. Probably a bug in how the Windows backend hacks around Windows' limitation to only store one password per site. Glad to hear the pending PR might address that. Would be nice to get a unit test to capture the expectation.

**Comment by jaraco** [comment]
Oh, man. I think it's worse than reported above. Now that I look at the implementation, it appears as if `get_password` [gives preference to the credential with the simple name](https://github.com/jaraco/keyring/blob/04269447ae7d30d3e31b9122fcd0e0358c87eb51/keyring/backends/Windows.py#L99), but `get_credential` gives [preference to the credential with the compound name](https://github.com/jaraco/keyring/blob/04269447ae7d30d3e31b9122fcd0e0358c87eb51/keyring/backends/Windows.py#L165).

That inconsistency wouldn't make a difference except in this case where duplicate entries are created.

**Comment by jrobbins-LiveData** [comment]
To preserve what I learned while working on PR #544, [set_password](https://github.com/jaraco/keyring/blob/04269447ae7d30d3e31b9122fcd0e0358c87eb51/keyring/backends/Windows.py#L118) doesn't seem to implement the intent expressed in the [comment ](https://github.com/jaraco/keyring/blob/04269447ae7d30d3e31b9122fcd0e0358c87eb51/keyring/backends/Windows.py#L76)

>     Passwords are stored under the service name unless there is a collision
>     (another password with the same service name but different user name),
>     in which case the previous password is moved into a compound name:
>     {username}@{service}

because `set_password` seems to ignore checking for a "different user name".

```
    def set_password(self, service, username, password):
        existing_pw = self._get_password(service)
        if existing_pw:
            # resave the existing password using a compound target
            existing_username = existing_pw['UserName']
            target = self._compound_name(existing_username, service)
            self._set_password(
                target,
                existing_username,
                existing_pw.value,
            )
        self._set_password(service, username, str(password))
```

In other words, `existing_username` should be compared with incoming `username`, and the "move" should only be done if the two usernames differ.  Fixing this would prevent spurious "moves". 

A unit test that expects no spurious moves should be feasible.  A unit test that shows the inconsistency pointed out [above ](https://github.com/jaraco/keyring/issues/545#issuecomment-981121347) should also be feasible.

**Comment by wl2776** [comment]
> Passwords are stored under the service name unless there is a collision
> (another password with the same service name but different user name),
> in which case the previous password is moved into a compound name:
> {username}@{service}

That wouldn't make a problem, if `get_credentials(server, None)` would retrieve the same password as was set with `set_password`.  Now it returns `{username}@{service}`.
